src/Components/Server/src/ComponentHub.cs

```diff
@@ -66,6 +66,7 @@ public override Task OnDisconnectedAsync(Exception exception)
             }
 
             CircuitHost = null;
+            circuitHost.UnhandledException -= CircuitHost_UnhandledException;
```

Comment:
This was a bug, nothing would unsubscribe, and as a result this could cause us to leak a Hub and it's related objects per-reconnection.

Reply:
Is this definitely right? Maybe I'm misreading it, but it looks like:

 * We attach the `UnhandledException` handler in `StartCircuit` (and *not* in `ConnectCircuit`)
 * We now detach the handler in `OnDisconnectedAsync`
 * If the client later reconnects, we don't re-attach the `UnhandledException` handler

More broadly, I don't follow in the existing code why we set `CircuitHost = null` during `OnDisconnectedAsync`. This means that once disconnected, the client could call `StartCircuit` again instead of `ConnectCircuit`, or if they called `BeginInvokeDotNetFromJS` after disconnection and before calling `ConnectCircuit` that they'd get an incorrect error message ("Circuit not initialized" instead of something like "Circuit not connected").

Reply:
> If the client later reconnects, we don't re-attach the UnhandledException handler, which would be a bug, right?

We do re-attach https://github.com/aspnet/AspNetCore/pull/12714/files#diff-9b2cc454c4a964ad53e0549df0fd4fc6R128

> Instead of this, I'd expect us to keep the handler live until the circuit host is disposed, so we can deal with exceptions even while disconnected.

If we want to do this then the fix is much more substantial. We don't want to have the circuit host hold on to an *old* instance of a the hub.

> More broadly, I don't follow in the existing code why we set CircuitHost = null during OnDisconnectedAsync. This means that once disconnected, the client could call StartCircuit again instead of ConnectCircuit, or if they called BeginInvokeDotNetFromJS after disconnection and before calling ConnectCircuit that they'd get an incorrect error message ("Circuit not initialized" instead of something like "Circuit not connected").

We're nulling out to `CircuitHost` to separate the host from the SignalR connection. I guess it doesn't super matter since the connection is now gone.  Do you see an angle where this is a problem?

Reply:
I talked with the SignalR folks - the associated between the `Items` (which we use to store the `CircuitHost`) is 1:1 with the connection.

Reply:
OK right, now I see where my comment was on the wrong track. I thought the object here would persist across the lifetime of the circuit (as the circuit host does). Now I realise this is the *hub* and that's one instance per reconnection. So I agree it's correct to tear down all this stuff on disconnect.

> We do re-attach

Ah yes. I was looking at the old code in my local copy and saw it didn't reattach during `ConnectCircuit`. I see now that your PR changes that, so we're OK.

> We're nulling out to CircuitHost to separate the host from the SignalR connection. I guess it doesn't super matter since the connection is now gone. Do you see an angle where this is a problem?

This seems fine now I understand which object we're talking about.

Reply:
I discussed this with @BrennanConroy as well. It's not really doing anything when we set the `CircutiHost` to null, because the items collection is tied to the lifetime of the connection. I'm just going to take that line out so we're not confused by it and add a comment elsewhere.

Reply:
Discused offline, I'm going to follow up on the unhandled exception part of this PR in my next one.